src/Clapi/Relay.hs

```diff
@@ -135,11 +152,14 @@ relay vs = waitThenFwdOnly fwd
               else qDefs'
             qContOps = Map.mapKeys (ns :</) contOps
             qContOps' = vsMinimiseContOps qContOps vs
-            qContOps'' = if shouldPubRoot
-              then Map.insert Root (nsContOp rootDef) qContOps'
-              else qContOps'
             mungedTas = Map.mapWithKey
               (\p tn -> (tn, either error id $ getLiberty p vs')) updatedTyAssns
+            getContOps p = case fromJust $ treeLookupNode p $ vsTree vs' of
```

Comment:
Hmm, interesting. Perhaps `Tree.hs` should have a "get me the parent node of this path" function, so that we don't have to do the error handing outside like this. (It's part of the guarantees that the `RoseTree` structure is trying to give us, so it's silly we have to assume it outside)

Reply:
Will do.

Reply:
Reading this again, that isn't what this is doing as such, the `fromJust` here is that we've just type assigned this path, so it should blinking well have some data. The one below is getting the path from the old tree (where it might not be) so that one should be able to fail.

Reply:
Ah, ok, I misunderstood that.